File: Homework/08:24:2022/PassingData_ObjC/PassingData_ObjC/ViewController.m
//
//  ViewController.m
//  PassingData_ObjC
//
//  Created by Consultant on 8/24/22.
//

#import "ViewController.h"
#import "ViewControllerB.h"
#import "ViewControllerC.h"

@interface ViewController ()

@end

@implementation ViewController

- (IBAction)propertiesButtonAction:(id)sender {
    UIStoryboard *board;

    if (!self.storyboard)
    {
        board = [UIStoryboard storyboardWithName:@"Main" bundle:nil];
    }
    else
    {
        board = self.storyboard;
    }

    ViewControllerB *vc = [board instantiateViewControllerWithIdentifier:@"viewControllerB"];
    //[self.navigationController pushViewController:View animated:YES];
    
    //ViewControllerB *secondViewController = [[ViewControllerB alloc] init];
    vc.data = @"Hi! I'm Luis"; // Set the exposed property
    [self.navigationController pushViewController:vc animated:YES];
}
- (IBAction)segueButtonAction:(id)sender {
    [self performSegueWithIdentifier:@"showDetailingSegue" sender:self];
}
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    if([segue.identifier isEqualToString:@"showDetailingSegue"]){
        ViewControllerC *controller = (ViewControllerC *)segue.destinationViewController;
        controller.data = @"Hi Segue";
    }
}

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
}


@end

File: Homework/08:24:2022/PassingData4-1/PassingData4-1/ViewController.swift
//
//  ViewController.swift
//  PassingData4-1
//
//  Created by Consultant on 8/24/22.
//

import UIKit

class ViewController: UIViewController {
    // MARK: - UI
    
    @IBOutlet weak var helloUILabel: UILabel!
    @IBOutlet weak var byeUILabel: UILabel!
    @IBOutlet weak var welcomeUILabel: UILabel!
    var hello = "Salut", bye = "Au revoir", welcome = "Bienvenue"
    @IBAction func englishButtonAction(_ sender: Any) {
        onButtonTap()
    }
    @IBAction func segueButtonAction(_ sender: Any) {
        onSegueTap()
    }
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        let segueDestination = segue.destination as! ViewControllerC
        segueDestination.message = "Hello World!!!"
    }
    override func viewDidLoad() {
        super.viewDidLoad()
        helloUILabel?.text = hello
        byeUILabel?.text = bye
        welcomeUILabel?.text = welcome
        // Do any additional setup after loading the view.
    }
    func onSegueTap(){
        self.performSegue(withIdentifier: "showDetailingSegue", sender: self)
    }
    func onButtonTap(){
        let vc =  storyboard?.instantiateViewController(withIdentifier: "viewControllerB") as! ViewControllerB
        vc.helloB = "Hello"
        vc.byeB = "Bye"
        vc.welcomeB = "Welcome"
        navigationController?.pushViewController(vc, animated: true)
        //present(vc, animated: true, completion: nil)
    }

}

